### ISSUE TITLE
Better workflow - release vs debug, caching, ...

### ISSUE THREAD

**Issue by zrzka** [resin-io-projects/resin-rust-hello-world#7]
I think that this example should be updated, because Rust popularity is growing. I'd add following things.

## Building

* dependencies caching (see [Dockerfile.template](https://github.com/zrzka/rust-thermometer/blob/master/Dockerfile.template))
* allow to select `debug` vs `--release`

## Sample scripts

* `local-push.sh`
    * generates `Dockerfile` from `Dockerfile.template`
    * pushes `debug` by default

## Final image

* do not include source code
* do not include build tools
* just final binary (+ assets, ...)
* again, see [Dockerfile.template](https://github.com/zrzka/rust-thermometer/blob/master/Dockerfile.template)

To sum it up, show users the complete workflow. Development -> production.

**Comment by zrzka** [comment]
https://github.com/resin-io-projects/resin-rust-hello-world/pull/8 merged.
